tests/integration/targets/aci_l3out_bgp_peer/tasks/main.yml

```diff
@@ -1500,6 +1501,159 @@
+      interface_profile: ansible_interface_profile
+      pod_id: 1
+      node_id: 201
+      path_ep: eth1/15
+      peer_ip: 192.168.50.0/24
+      remote_asn: 65456
+      bgp_controls:
+        - nh-self
+        - send-com
+        - send-ext-com
+      peer_controls:
+        - bfd
+      address_type_controls:
+        - af-ucast
+      ttl: 2
+      local_as_number_config: "replace-as"
+      state: present
+    register: add_dyn_bgp_peer
+
+  - name: verify dynamic BGP peer (prefix) has been created with correct attributes
+    ansible.builtin.assert:
+      that:
+        - add_dyn_bgp_peer.current.0.bgpPeerP.attributes.dn == "uni/tn-ansible_tenant/out-ansible_l3out/lnodep-ansible_node_profile/lifp-ansible_interface_profile/rspathL3OutAtt-[topology/pod-1/paths-201/pathep-[eth1/15]]/peerP-[192.168.50.0/24]"
+        - add_dyn_bgp_peer.current.0.bgpPeerP.attributes.addr == "192.168.50.0/24"
+        - add_dyn_bgp_peer.current.0.bgpPeerP.attributes.addrTCtrl == "af-ucast"
+        - add_dyn_bgp_peer.current.0.bgpPeerP.attributes.adminSt == "enabled"
+        - add_dyn_bgp_peer.current.0.bgpPeerP.attributes.ctrl == "nh-self,send-com,send-ext-com"
+        - add_dyn_bgp_peer.current.0.bgpPeerP.attributes.peerCtrl == "bfd"
+        - add_dyn_bgp_peer.current.0.bgpPeerP.attributes.ttl == "2"
+        - add_dyn_bgp_peer.current.0.bgpPeerP.attributes.annotation == 'orchestrator:ansible'
+        - add_dyn_bgp_peer.current.0.bgpPeerP.children.0.bgpLocalAsnP.attributes.asnPropagate == "replace-as"
+        - add_dyn_bgp_peer.current.0.bgpPeerP.children.0.bgpLocalAsnP.attributes.localAsn == "0"
+
+  - name: verify remote AS object has been created correctly
+    ansible.builtin.assert:
+      that:
+        - add_dyn_bgp_peer.current.0.bgpPeerP.children.2.bgpAsP.attributes.asn == "65456"
+
+  # ADD dynamic BGP peer (prefix) again to check idempotence
+  - name: add dynamic BGP peer (prefix) to ethernet port again
+    cisco.aci.aci_l3out_bgp_peer:
+      <<: *dyn_bgp_peer
+    register: add_dyn_bgp_peer_again
+
+  - name: verify dynamic BGP peer (prefix) has been created with correct attributes
+    ansible.builtin.assert:
+      that:
+        - add_dyn_bgp_peer_again is not changed
+        - add_dyn_bgp_peer_again.current.0.bgpPeerP.attributes.dn == "uni/tn-ansible_tenant/out-ansible_l3out/lnodep-ansible_node_profile/lifp-ansible_interface_profile/rspathL3OutAtt-[topology/pod-1/paths-201/pathep-[eth1/15]]/peerP-[192.168.50.0/24]"
+        - add_dyn_bgp_peer_again.current.0.bgpPeerP.attributes.addr == "192.168.50.0/24"
+        - add_dyn_bgp_peer_again.current.0.bgpPeerP.attributes.addrTCtrl == "af-ucast"
+        - add_dyn_bgp_peer_again.current.0.bgpPeerP.attributes.adminSt == "enabled"
+        - add_dyn_bgp_peer_again.current.0.bgpPeerP.attributes.ctrl == "nh-self,send-com,send-ext-com"
+        - add_dyn_bgp_peer_again.current.0.bgpPeerP.attributes.peerCtrl == "bfd"
+        - add_dyn_bgp_peer_again.current.0.bgpPeerP.attributes.ttl == "2"
```

Comment:
is there a reason for not asserting the children here?

Reply:
Thanks, yeah think the new parts we should just address right away. Bit tedious perhaps but then at least it is done better this time. I do not expect you to fix all the other tests since this was also not introduced by you ( feel free if you feel the urge to improve as it is much appreciated).